frontend/src/app/tasks/components/TaskBox.tsx

```diff
@@ -4,21 +4,32 @@ import { useState } from 'react'
 import Box from '@mui/joy/Box'
 import Button from '@mui/joy/Button'
 import Typography from '@mui/joy/Typography'
-import { Delete24Filled } from '@fluentui/react-icons'
+import { Delete24Filled, Edit24Filled } from '@fluentui/react-icons'
+
+import { Project } from '@/domain/Project'
+import { TaskType } from '@/domain/TaskType'
 
 import { ScreenReaderOnly } from '@/ui/ScreenReaderOnly/ScreenReaderOnly'
 import { ConfirmationModal } from '@/ui/ConfirmationModal/ConfirmationModal'
 import { styled } from '@mui/joy/styles'
 import { Task } from '@/domain/Task'
 import { getTimeDifference, convertTimeToMinutes } from '../utils/time'
 
+import { EditTask } from './EditTask'
+import { useDeleteTask } from '../hooks/useTask'
+
 type TaskProps = {
   task: Task
-  deleteTask: (taskId: number) => void
+  tasks: Array<Task>
```

Comment:
If this component is meant to be a single task, then why are we passing in the whole array of tasks? Is it so we can check overlap with other tasks?

Reply:
Yes! its basically for this function:
```ts

() => {
    const validation = Task.safeParse(formState)

    if (!validation.success) {
      validation.error.issues.map(({ message }) => {
        showError(message)
      })

      return
    }

    const { message } = getOverlappingTasks(
      formState,
      tasks.filter(({ id }) => id !== formState.id)
    )

    if (message.length > 0) {
      showError(message)
      return
    }

    editTask(formState)
```